src/bootstrap/tool.rs

```diff
@@ -652,20 +653,20 @@ macro_rules! tool_extended {
 // Note: tools need to be also added to `Builder::get_step_descriptions` in `build.rs`
 // to make `./x.py build <tool>` work.
 tool_extended!((self, builder),
-    Cargofmt, rustfmt, "src/tools/rustfmt", "cargo-fmt", {};
-    CargoClippy, clippy, "src/tools/clippy", "cargo-clippy", {};
-    Clippy, clippy, "src/tools/clippy", "clippy-driver", {};
-    Miri, miri, "src/tools/miri", "miri", {};
-    CargoMiri, miri, "src/tools/miri/cargo-miri", "cargo-miri", {};
-    Rls, rls, "src/tools/rls", "rls", {
+    Cargofmt, rustfmt, "src/tools/rustfmt", "cargo-fmt", default=true, {};
+    CargoClippy, clippy, "src/tools/clippy", "cargo-clippy", default=true, {};
+    Clippy, clippy, "src/tools/clippy", "clippy-driver", default=true, {};
+    Miri, miri, "src/tools/miri", "miri", default=false, {};
```

Comment:
This will disable miri on nightly and dev builds.

Something like that should enable miri for nightly and dev builds:
```rust
    Miri, miri, "src/tools/miri", "miri", default=builder.build.unstable_features(), {};
```
Alternatively enable for nightly only:
```rust
    Miri, miri, "src/tools/miri", "miri", default=builder.build.config.channel == "nightly", {};
```

Reply:
I can't use run-time values though if I put this into the `DEFAULT` constant (whatever that does). Also I think hygiene will prevent me from referring to `builder`, but I could use a closure to get around that.

Reply:
Right, maybe Mark comes up with an idea.

Reply:
For now I changed the macro to say whether a tool is "stable", and then just enable all tools per default on nightly/dev builds.

But I still have no idea what to do with that undocumented `DEFAULT` constant.